src/uct/ib/ud/base/ud_ep.c

```diff
@@ -596,9 +603,13 @@ ucs_status_t uct_ud_ep_create_connected_common(const uct_ep_params_t *ep_params,
         goto out;
     }
 
+    /* Remove RX and CONNECT_TO_EP flags, since the endpoint is created as full
+     * duplex and p2p by default, but this endpoint should be TX-only */
+    ep->flags &= ~(UCT_UD_EP_FLAG_CONNECT_TO_EP | UCT_UD_EP_FLAG_RX);
```

Comment:
maybe pass the right flags to uct_ud_ep_create, instead of adjusting them later?